src/views/bootsources/utils/utils.ts

```diff
@@ -0,0 +1,46 @@
+import useInstanceTypesAndPreferences from '@catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/hooks/useInstanceTypesAndPreferences';
+import {
+  DEFAULT_INSTANCETYPE_LABEL,
+  DEFAULT_PREFERENCE_LABEL,
+} from '@catalog/CreateVMHorizontalNav/components/CreateFromInstanceTypes/utils/constants';
+import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { V1alpha2VirtualMachineClusterPreference } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { DataSourceModelGroupVersionKind } from '@kubevirt-utils/models';
+import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import {
+  K8sResourceCommon,
+  Operator,
+  useK8sWatchResource,
+  WatchK8sResult,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+export const useBootableVolumes = (): WatchK8sResult<V1beta1DataSource[]> =>
+  useK8sWatchResource<V1beta1DataSource[]>({
+    groupVersionKind: DataSourceModelGroupVersionKind,
+    isList: true,
+    selector: {
+      matchExpressions: [
+        { key: DEFAULT_PREFERENCE_LABEL, operator: Operator.Exists },
+        { key: DEFAULT_INSTANCETYPE_LABEL, operator: Operator.Exists },
+      ],
+    },
+  });
+
+export const getDataSourcePreferenceLabelValue = (
+  obj: V1beta1DataSource | K8sResourceCommon,
+): string => obj?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL];
+
+export const useClusterPreference = (
+  obj: V1beta1DataSource,
+): V1alpha2VirtualMachineClusterPreference => {
+  const preferenceLabelValue = getDataSourcePreferenceLabelValue(obj);
+  const { preferences } = useInstanceTypesAndPreferences();
+
+  return preferences?.find((preference) => preference?.metadata?.name === preferenceLabelValue);
+};
+
+export const usePreferenceReadableOS = (obj: V1beta1DataSource): string => {
```

Comment:
perfermence nit suggestion:
calling this hook in each line will cause a lot of computations that can be avoided,
for example, you can fetch the preferences at the table scope, create an object that would look like: `[obj.metadata.name]: obj`, and send that object as a prop to the row, than you can reach the preference with just calling object[obj.metadata.name] and you did the entire computation just once at table scope

Reply:
I am aware of this. I just haven't figured out how to improve this. Because when dealing with that, I was hitting some other issues.